File: pact-jvm-matchers/src/main/scala/au/com/dius/pact/matchers/MatchingConfig.scala
package au.com.dius.pact.matchers

import scala.collection.mutable

object MatchingConfig {
  var bodyMatchers = mutable.HashMap[String, BodyMatcher](
    "application/.*xml" -> new XmlBodyMatcher(),
    "application/.*json" -> new JsonBodyMatcher(),
    "application/json-rpc" -> new JsonBodyMatcher(),
    "application/jsonrequest" -> new JsonBodyMatcher()
  )

  def lookupBodyMatcher(mimeType: String): Option[(String, BodyMatcher)] = {
    bodyMatchers.find(entry => mimeType.matches(entry._1))
  }
}

File: pact-jvm-matchers/src/main/kotlin/au/com/dius/pact/matchers/MatchingConfig.kt
package au.com.dius.pact.matchers

object MatchingConfig {
  val bodyMatchers = mapOf(
    "application/.*xml" to "au.com.dius.pact.matchers.XmlBodyMatcher",
    "text/xml" to "au.com.dius.pact.matchers.XmlBodyMatcher",
    "application/.*json" to "au.com.dius.pact.matchers.JsonBodyMatcher",
    "application/json-rpc" to "au.com.dius.pact.matchers.JsonBodyMatcher",
    "application/jsonrequest" to "au.com.dius.pact.matchers.JsonBodyMatcher",
    "text/plain" to "au.com.dius.pact.matchers.PlainTextBodyMatcher",
    "multipart/form-data" to "au.com.dius.pact.matchers.MultipartFormBodyMatcher",
    "application/x-www-form-urlencoded" to "au.com.dius.pact.matchers.FormPostBodyMatcher"
  )

  @JvmStatic
  fun lookupBodyMatcher(mimeType: String): BodyMatcher? {
    val matcher = bodyMatchers.entries.find { mimeType.matches(Regex(it.key)) }?.value
    return if (matcher != null) {
      Class.forName(matcher)?.newInstance() as BodyMatcher?
    } else {
      null
    }
  }
}
